geode-core/src/test/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderQueueJUnitTest.java

```diff
@@ -201,18 +306,40 @@ private BucketRegionQueue mockBucketRegionQueue(final Queue backingList) {
     // mocked pr region
     when(mockBucketRegion.getLocalMaxMemory()).thenReturn(100);
     when(mockBucketRegion.size()).thenReturn(backingList.size());
-
     BucketRegionQueue bucketRegionQueue = mock(BucketRegionQueue.class);
     when(bucketRegionQueue.getPartitionedRegion()).thenReturn(mockBucketRegion);
-    when(bucketRegionQueue.peek()).thenAnswer((Answer) invocation -> backingList.poll());
+    when(bucketRegionQueue.peek())
+        .thenAnswer((Answer) invocation -> pollAndWaitIfNull(backingList));
+    when(bucketRegionQueue.getElementsMatching(any(), any()))
+        .thenAnswer((Answer) invocation -> Arrays
+            .asList(new Object[] {getFirstNotNull(backingList)}));
     return bucketRegionQueue;
   }
 
+  private Object pollAndWaitIfNull(Queue queue) {
+    Object object = queue.poll();
+    if (object == null) {
+      try {
+        Thread.sleep(10);
+      } catch (InterruptedException e) {
+        e.printStackTrace();
+      }
+    }
```

Comment:
It is a way to simulate some delay in the arrival of events to the queue. If a null element is found in the queue, some delay is added to the polling execution in order to be used in cases where the batch must be finished, not because the max size was reached but because the peeking took longer than the timeToWait passed.